src/micromamba/install.sh

```diff
@@ -104,19 +45,71 @@ install_micromamba() {
     fi
     url="https://micro.mamba.pm/api/micromamba/linux-${arch}/${version}"
 
-    check_packages curl ca-certificates bzip2
-    echo "Downloading micromamba..."
-    curl -sL "${url}" | tar -xj -C /usr/local/bin/ --strip-components=1 bin/micromamba
+    echo "Installing prerequisites for downloading micromamba..."
+    ensure_download_prerequisites
+    echo "Downloading micromamba from ${url}..."
+    download_with_curl "${url}" "${destination}"
+    echo "Micromamba download complete."
 }
 
+run_as_user() {
+    su "${USERNAME}" "${@}"
+}
+
+micromamba_as_user() {
+    run_as_user bash -c "micromamba $*"
+}    
+
 export DEBIAN_FRONTEND=noninteractive
 
-# Soft version matching
-find_version_from_git_tags VERSION "https://github.com/mamba-org/mamba" "tags/micromamba-"
+ensure_path_for_login_shells
+
+if [ "${REINSTALL}" = "false" ]; then
+    if type micromamba > /dev/null 2>&1; then
+        echo "Detected existing micromamba: $(micromamba --version)."
+        echo "The reinstall argument is false, so not overwriting."
+        skip_install="true"
+    fi
+fi
+
+if [ "${skip_install}" != "true" ]; then
+    install_micromamba "${VERSION}" "${micromamba_destination}"
+fi
+
+add_conda_group() {
+    if ! cat /etc/group | grep -e "^conda:" > /dev/null 2>&1; then
+        groupadd -r conda
+    fi
+}
 
-install_micromamba "${VERSION##*-}"
+initialize_root_prefix() {
+    mkdir -p "${MAMBA_ROOT_PREFIX}/conda-meta"
+    touch "${MAMBA_ROOT_PREFIX}/conda-meta/history"
+    add_conda_group
+    usermod -a -G conda "${USERNAME}"
+    chown -R "${USERNAME}:conda" "${MAMBA_ROOT_PREFIX}"
+    chmod -R g+r+w "${MAMBA_ROOT_PREFIX}"
+    find "${MAMBA_ROOT_PREFIX}" -type d -print0 | xargs -n 1 -0 chmod g+s
+}
 
-# Clean up
-rm -rf /var/lib/apt/lists/*
+initialize_root_prefix
+
+if [ "${ADD_CONDA_FORGE}" = "true" ]; then
+    echo "Appending 'conda-forge' to channels"
+    micromamba_as_user config append channels conda-forge
+fi
+
+echo "Setting channel_priority to strict"
+micromamba_as_user config set channel_priority strict
+
+echo "Initializing Bash shell"
+micromamba_as_user shell init --shell=bash
+su -c "if ! grep -q 'micromamba activate # added by micromamba devcontainer feature' ~/.bashrc; then echo 'micromamba activate # added by micromamba devcontainer feature' >> ~/.bashrc; fi" - "${USERNAME}"
+
+if type zsh > /dev/null 2>&1; then
+    echo "Initializing zsh shell"
+    micromamba_as_user shell init --shell=zsh
+    su -c "if ! grep -q 'micromamba activate # added by micromamba devcontainer feature' ~/.zshrc; then echo 'micromamba activate # added by micromamba devcontainer feature' >> ~/.zshrc; fi" - "${USERNAME}"
+fi
 
-echo "Done!"
+echo "Done installing micromamba!"
```

Comment:
Since the Reference Features are "Done!" only, I think it would be more consistent to leave this as it is.